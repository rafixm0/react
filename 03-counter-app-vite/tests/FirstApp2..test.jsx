const { render, screen } = require("@testing-library/react");
const { FirstApp } = require("../src/FirstApp");


describe('Pruebas en FirstApp />', () => {

    test('Debe de hacer match con el snapshot', () => {

        const title = 'Hola, soy Vegeta';

        const { container } = render( <FirstApp title={title} />);

        expect( container ).toMatchSnapshot();

    })

    test('Debe de mostrar el mensaje: "Hola, soy Vegeta"', () => {

        const title = 'Hola, soy Vegeta';

        render( <FirstApp title={title} />);
        //screen.debug();
        expect( screen.getByText(title) ).toBeTruthy();

    })


    test('Debe mostrar el título en un <h1>', () => {

        const title = 'Rafa';

        render( <FirstApp title={title} />);
        expect( screen.getByRole('heading', {level: 1}).innerHTML ).toContain( title );

    })

    test('Debe de mostrar el subtítulo enviado por props', () => {

        const title = 'Rafa';
        const subTitle = 'Soy el mismo OBJ';

        render( <FirstApp title={title} subtitle={subTitle} />);
        expect( screen.getAllByText(subTitle).length).toBe(1);

    })


})