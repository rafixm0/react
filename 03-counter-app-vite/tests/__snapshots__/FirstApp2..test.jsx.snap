// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp /> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    data-testid="test-title"
  >
    Hola, soy Vegeta
  </div>
  <h1>
    Rafa
  </h1>
  <p>
    Soy el mismo OBJ
  </p>
</div>
`;
